src/fromJCAMP.js

```diff
@@ -12,59 +13,145 @@ export function fromJCAMP(buffer) {
     keepRecordsRegExp: /.*/,
     profiling: true,
   });
+
   let dataStructure = [];
-  let entries = parsedData.flatten;
+  let entries = parsedData.entries;
+
   for (let entry of entries) {
-    if ((entry.spectra && entry.spectra.length > 0) || entry.minMax) {
-      let metadata = Object.assign({}, entry.info, entry.meta);
-      let info = getInfoFromJCAMP(metadata);
-
-      if (info.experiment === 'wobble_curve') continue;
-
-      let dimensions = [];
-      let dependentVariables = [];
-
-      let dependentVariable = {};
-      if (info.dimension === 1) {
-        for (let i = 0; i < entry.spectra.length; i++) {
-          let data = entry.spectra[i].data;
-          data = convertToFloatArray(data);
-        }
-        dependentVariable.components = entry.spectra;
-      } else if (info.dimension === 2) {
-        entry.minMax.z = convertToFloatArray(entry.minMax.z);
-        dependentVariable.components = entry.minMax;
-      }
-      let dimension = {
-        increment: info.increment,
-        numberOfPoints: info.numberOfPoints,
-      };
+    const { dataType = '', dataClass = '' } = entry;
+    const processor = chooseProcessor(dataType, dataClass);
+    if (!processor) continue;
+    processor(entry, dataStructure);
+  }
+  return dataStructure.map((_data) => {
+    if (_data.spectra) return _data.spectra[0];
+    return _data;
+  });
+}
 
-      if (info.fid) {
-        dimension.coordinatesOffset = {
-          magnitude: -info.digitalFilter * info.increment,
-          units: 'second',
-        };
-      } else {
-        dimension.coordinatesOffset = {
-          magnitude:
-            info.frequencyOffset / info.baseFrequency - 0.5 * info.spectraWidth,
-          units: 'ppm',
-        };
+function chooseProcessor(dataType, dataClass) {
+  switch (dataType.toLowerCase()) {
+    case 'link':
+      return extractFromLink;
```

Comment:
What is it expected to be done when it is a link ? We should just do nothing no ?